### DISCUSSION TITLE
Processing a list in multiple ways, in one command

### DISCUSSION THREAD

**Discussion by BatmanAoD** [nushell/nushell#10527] [Q&A] [answered]
I want to do something like this:

```
<generate a list> | $"(math min), (math max), (math avg)"
```

to calculate the min, max, and average from a list. The command above doesn't work because there's no value actually piped in to the `math` command. (I don't actually need the output to be a string; generating a table would be even better.)

I found out how to do this with an intermediate variable:

```
let listvar = (<generate>)
$"($listvar | math min), ($listvar | math max)"
```

Is there a way to do this all in one command?

**Comment by BatmanAoD** [comment]
...actually, in my case, it's not a list but a single-column table. I'm not sure if that makes a difference.

**Reply by amtoine** [reply, marked as answer]
@BatmanAoD 
you can do that by using the `$in` builtin variable :yum:

- with a `list`
```nushell
> seq 1 10 | $"($in | math min), ($in | math max), ($in | math avg)"
1, 10, 5.5
```
- with one of the columns of a table
```nushell
> ls | get size | $"($in | math min), ($in | math max), ($in | math avg)"
921 B, 41.5 KB, 6.8 KB
```

**Reply by BatmanAoD** [reply]
Ah, `$in |` is exactly what I needed, thanks!
